### PR TITLE
refactor: session launcher image parsing

### PR DESCRIPTION
**Pull request by agatha197**
**Changes:**
This PR enhances the image selection UI to support extended image information in the session launcher, providing more detailed and structured image metadata display.

Key updates:
- Adds support for displaying base image name, version, architecture and tags in a more organized format
- Introduces tag aliasing functionality to show more user-friendly tag names
- Updates the image selection interface to show customized image tags with distinct styling
- Adds translations for "Tags" across all supported languages

**Minimum required manager version:** 24.09.1

**Review Requirements:**
1. Verify extended image info display works when backend supports 'extended-image-info' feature
2. Check tag aliasing correctly translates technical tags to user-friendly names
3. Confirm customized image tags are properly highlighted with cyan color
4. Validate all translations for "Tags" appear correctly across languages

**Test Cases:**
1. Select an image with extended info:
   - Should show base image name, version, architecture and tags
   - Tags should display with proper aliasing
2. Select a customized image:
   - Should show customized tag with cyan highlighting
   - Should display proper customized image name from labels
3. Test with extended-image-info feature disabled:
   - Should fallback to legacy image info display format
4. Test the image of `Environments & Resource allocation` page and `Confirm and Launch` page has the same value.

**Screenshots:**

### CODE REVIEW

**Comment by agatha197** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2800?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2800** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2800?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2795** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2795?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2794** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2794?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2796** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2796?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2788** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2788?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2789** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2789?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2787** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2787?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2785** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2785?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @agatha197 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/ResourceAllocationFormItems.test.ts [hunk 1]
```diff
@@ -46,6 +46,9 @@ describe('getAllocatablePresetNames', () => {
     registry: 'registry1',
     tag: 'tag1',
     resource_limits: [{ key: 'cuda.shares', min: '1', max: '1' }],
+    base_image_name: undefined,
+    tags: undefined,
+    version: undefined,
```
Comment (agatha197) [inline react/src/components/ResourceAllocationFormItems.test.ts:51]: Is it okay to do it like this? @yomybaby

File: react/src/pages/SessionLauncherPage.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import BAICard from '../BAICard';
 import BAIIntervalText from '../components/BAIIntervalText';
+import CopyableCodeText from '../components/CopyableCodeText';
```
Comment (ironAiken2) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion

```

File: react/src/pages/SessionLauncherPage.tsx [hunk 3]
```diff
@@ -36,12 +37,15 @@ import SourceCodeViewer from '../components/SourceCodeViewer';
 import VFolderTableFormItem, {
   VFolderTableFormValues,
 } from '../components/VFolderTableFormItem';
+import CopyButton from '../components/lablupTalkativotUI/CopyButton';
```
Comment (ironAiken2) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion

```

File: react/src/pages/SessionLauncherPage.tsx [hunk 4]
```diff
@@ -1295,67 +1304,190 @@ const SessionLauncherPage = () => {
...
-                                  >
-                                    {form.getFieldValue('environments')?.manual}
-                                  </Typography.Text>
-                                ) : (
-                                  <>
-                                    <SessionKernelTags
-                                      image={
+                          {supportExtendedImageInfo ? (
+                            <Row
+                              style={{ flexFlow: 'nowrap', gap: token.sizeXS }}
```
Comment (ironAiken2) [inline react/src/pages/SessionLauncherPage.tsx]: It's super minor thing, but looks a little far between image and string
```suggestion
                              style={{ flexFlow: 'nowrap' }}
```

File: react/src/pages/SessionLauncherPage.tsx [hunk 5]
```diff
@@ -1295,67 +1302,186 @@ const SessionLauncherPage = () => {
...
-                                        ?.manual,
-                                    }}
-                                  >
-                                    {form.getFieldValue('environments')?.manual}
-                                  </Typography.Text>
-                                ) : (
-                                  <>
-                                    <SessionKernelTags
-                                      image={
+                          {supportExtendedImageInfo ? (
```
Comment (ironAiken2) [inline react/src/pages/SessionLauncherPage.tsx:1305]: I think it used to scroll when there were more tags, but now the text looks broken.

File: react/src/pages/SessionLauncherPage.tsx [hunk 6]
```diff
@@ -1295,67 +1302,186 @@ const SessionLauncherPage = () => {
...
+                            <Row style={{ flexFlow: 'nowrap' }}>
+                              <Col>
```
Comment (ironAiken2) [inline react/src/pages/SessionLauncherPage.tsx:1307]: Is there a reason why you used Row and Col?

File: react/src/components/ImageEnvironmentSelectFormItems.tsx [hunk 7]
```diff
@@ -695,39 +716,98 @@ const ImageEnvironmentSelectFormItems: React.FC<
...
+                                {metadataTagAlias}
+                              </TextHighlighter>
+                              <Divider type="vertical" />
+                              <TextHighlighter keyword={versionSearch}>
+                                {image?.architecture}
+                              </TextHighlighter>
+                            </Flex>
+                            <Flex
+                              direction="row"
+                              // set specific class name to handle flex wrap using css
```
Comment (ironAiken2) [inline react/src/components/ImageEnvironmentSelectFormItems.tsx:798]: (dismissed)

File: react/src/pages/SessionLauncherPage.tsx (hunk 6 above)
Comment (agatha197) [reply react/src/pages/SessionLauncherPage.tsx:1307]: For the manual image name. If the manual image is too long, the text in the manual image is overflowing the card.

File: react/src/components/ImageEnvironmentSelectFormItems.tsx (hunk 7 above)
Comment (agatha197) [reply react/src/components/ImageEnvironmentSelectFormItems.tsx:798]: what did you mean?

File: react/src/components/ResourceAllocationFormItems.test.ts (hunk 1 above)
Comment (yomybaby) [reply react/src/components/ResourceAllocationFormItems.test.ts:51]: Yes. This is kind of mocking. So this mocked image is for > v24.09.1.

**Review by yomybaby — APPROVED** [review]
LGTM

**Pull request merged**